modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3RequestRetryStats.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.repositories.s3;
+
+import com.amazonaws.Request;
+import com.amazonaws.util.AWSRequestMetrics;
+import com.amazonaws.util.TimingInfo;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.logging.ESLogMessage;
+import org.elasticsearch.common.util.Maps;
+
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicLongArray;
+
+/**
+ * This class emit aws s3 metrics as logs until we have a proper apm integration
+ */
+public class S3RequestRetryStats {
+
+    private static final Logger logger = LogManager.getLogger(S3RequestRetryStats.class);
+
+    private final AtomicLong requests = new AtomicLong();
+    private final AtomicLong exceptions = new AtomicLong();
+    private final AtomicLong throttles = new AtomicLong();
```

Comment:
I am curious why you changed these from `LongAdder` to `AtomicLong`. Is it for better summation accuracy?

Reply:
No specific reason, just to keep it consistent with AtomicLongArray